flexget/api/core/database.py

```diff
@@ -17,53 +17,53 @@
 class ObjectsContainer(object):
     plugin_list = {'type': 'array', 'items': {'type': 'string'}}
 
+    database_input_object = {
+        'type': 'object',
+        'properties': {
+            'operation': {'type': 'string', 'enum': ['cleanup', 'vacuum', 'list_plugins', 'plugin_reset']}
+        },
+        'required': ['operation_name'],
+        'additionalProperties': False
+    }
 
-plugins_schema = api.schema('plugins_list', ObjectsContainer.plugin_list)
+    reset_plugin_input = database_input_object.copy()
+    reset_plugin_input['properties']['plugin_name'] = {'type': 'string'}
 
 
-@db_api.route('/cleanup/')
-class DBCleanup(APIResource):
-    @api.response(200, model=base_message_schema)
-    def post(self, session=None):
-        """ Make all plugins clean un-needed data from the database """
-        self.manager.db_cleanup(force=True)
-        return success_response('DB Cleanup finished')
+plugins_schema = api.schema('plugins_list', ObjectsContainer.plugin_list)
+input_schema = api.schema('db_schema', ObjectsContainer.database_input_object)
+reset_plugin_input_schema = api.schema('db_schema', ObjectsContainer.reset_plugin_input)
 
 
-@db_api.route('/vacuum/')
-class DBVacuum(APIResource):
+@db_api.route('/')
+class DBOperation(APIResource):
     @api.response(200, model=base_message_schema)
     def post(self, session=None):
-        """ Potentially increase performance and decrease database size"""
-        session.execute('VACUUM')
-        session.commit()
-        return success_response('DB VACUUM finished')
+        data = request.json
+        operation = data['operation']
+        if operation == 'cleanup':
+            self.manager.db_cleanup(force=True)
+            msg = 'DB Cleanup finished'
+        elif operation == 'vacuum':
+            session.execute('VACUUM')
+            session.commit()
+            msg = 'DB VACUUM finished'
+        elif operation == 'list_plugins':
```

Comment:
I think `list_plugins` should still be a GET request. It's idempotent and semantically that makes more sense.